linkerd/meshtls/rustls/src/creds.rs

```diff
@@ -27,19 +27,22 @@ pub fn watch(
     roots_pem: &str,
 ) -> Result<(Store, Receiver)> {
     let mut roots = rustls::RootCertStore::empty();
-    let certs = match rustls_pemfile::certs(&mut std::io::Cursor::new(roots_pem)) {
-        Err(error) => {
-            warn!(%error, "invalid trust anchors file");
-            return Err(error.into());
-        }
-        Ok(certs) if certs.is_empty() => {
-            warn!("no valid certs in trust anchors file");
-            return Err("no trust roots in PEM file".into());
-        }
-        Ok(certs) => certs,
-    };
+    let mut certs = vec![];
+    for cert in rustls_pemfile::certs(&mut std::io::Cursor::new(roots_pem)) {
+        match cert {
+            Err(error) => {
+                warn!(%error, "invalid trust anchors file");
+                return Err(error.into());
+            }
+            Ok(cert) => certs.push(cert),
+        };
+    }
+    if certs.is_empty() {
+        warn!("no valid certs in trust anchors file");
+        return Err("no trust roots in PEM file".into());
+    }
```

Comment:
In the interest of minimizing the diff, we can keep this more consistent with the pre-existing code:
```suggestion
    let certs = match rustls_pemfile::certs(&mut std::io::Cursor::new(roots_pem))
        .collect::<Result<Vec<_>, _>>()
    {
        Err(error) => {
            warn!(%error, "invalid trust anchors file");
            return Err(error.into());
        }
        Ok(certs) if certs.is_empty() => {
            warn!("no valid certs in trust anchors file");
            return Err("no trust roots in PEM file".into());
        }
        Ok(certs) => certs,
    };
```

Reply:
Done